src/1_api.js

```diff
@@ -191,10 +194,31 @@ Server.prototype.request = function(resource, data, storage, callback) {
 		url = u.url;
 		postData = u.data;
 	}
-	if (storage['getItem']('use_jsonp') || resource.jsonp) {
-		this.jsonpRequest(url, data, resource.method, callback);
-	}
-	else {
-		this.XHRRequest(url, postData, resource.method, storage, callback);
-	}
+
+	// How many times to retry the request if the initial attempt fails
+	var retries = 2;
+	// If request fails, retry after X miliseconds
+	/***
+	 * @type {function(?Error,*=): ?undefined}
+	 */
+	var done = function(err, data) {
+		if (err && retries > 0) {
+			retries--;
+			window.setTimeout(function() {
+				makeRequest();
+			}, 200);
```

Comment:
hmm, a few things:

1. we should count 5xx errors as failures, but not 4xx errors – we may have to fiugre out a smart way to do that
2. this doesn't actually set a timeout on requests, this jsut sets a request after a certain amount of time; it would be great to actually set a timeout on requests - otherwise if we fire off an JSONP request and it never returns, nothing will happen
3. let's move the constants (retires = 2, 200) as all-caps constants at the top of the page